### PR TITLE
Restructure wallet docs to prep for app wallet content

### PR DESCRIPTION
**Pull request by danpaul000**
#### Problem
Wallet creation documentation is inside the CLI guide.  In advance of full support by web wallets, we should have a top level Wallet Guide section.

#### Summary of Changes
Re-structure location of wallet docs, and move the Install CLI page inside the CLI guide section.

### CODE REVIEW

**Comment by danpaul000** [comment]
@garious @t-nelson structure only changes here, with minimal content change.  Will restore the content changes from my previous PR into the `wallet/` section once we agree on structure.

File: docs/src/SUMMARY.md [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 * [Introduction](introduction.md)
-* [Install the Solana Tool Suite](install-solana.md)
-* [Command-line Guide](cli/README.md)
-  * [Choose a Wallet](cli/choose-a-wallet.md)
-  * [Hardware Wallets](remote-wallet/README.md)
-    * [Ledger Hardware Wallet](remote-wallet/ledger.md)
-  * [Paper Wallet](paper-wallet/README.md)
-    * [Paper Wallet Usage](paper-wallet/usage.md)
+* [Wallet Guide](wallet/README.md)
+  * [Web Wallets](wallet/web-wallets.md)
```
Comment (garious) [inline docs/src/SUMMARY.md]: Why Web Wallets and not App Wallets?  Uber and Lyft, for example, are now commonly referred to as "App-based Rides".

File: docs/src/SUMMARY.md [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+  * [Command Line-based Wallets](wallet/cli-wallets.md)
```
Comment (garious) [inline docs/src/SUMMARY.md]: This can be "Command-line Wallets".  They're not line-based.

File: docs/src/wallet/cli-wallets.md [hunk 3]
```diff
@@ -1,11 +1,13 @@
-# Choose a Wallet
-
-Keypairs are stored in *wallets* and wallets come in many forms. A wallet might
-be a directory in your computer's file system, a piece of paper, or a
-specialized device called a *hardware wallet*. Some wallets are easier to use
-than others.  Some are more secure than others. In this section, we will compare
-and contrast different types of wallets and help you choose the wallet that
-best fits your needs.
+# Command Line Based Wallets
```
Comment (garious) [inline docs/src/wallet/cli-wallets.md]: same here

**Pull request merged**